packages/xstate-react/test/useMachine.test.tsx

```diff
@@ -945,4 +945,40 @@ describeEachReactMode('useMachine (%s)', ({ suiteKey, render }) => {
 
     expect(container.textContent).toBe('2');
   });
+
+  it('should deliver messages sent from an effect to an actor registered in the system', () => {
```

Comment:
this one was already working because we don't rehydrate deep children, so since the child was recreated anyway - it had a chance to register itself in the system